### ISSUE TITLE
MultiOrgan benchmark

### ISSUE THREAD

**Issue by Luugaaa** [ivadomed/contrast_agnostic_framework#15]
Datasets: 
* CHAOS - MRI: kidneys, spleen, liver) CT: liver
* SLIVER07: CT liver
* AMOS: multi organ 

Choas test set results: 

Preliminary results (only fold 0): 

<img width="1262" height="1106" alt="Image" src="https://github.com/user-attachments/assets/de3b2e71-d8f3-4ab6-be1f-b6e1bc0f2f28" />

When comparing v26_6_2_train050_val000 and v26_6_2_50_100, it seems using synthetic validation yields better results overall. 

Currently, our best approach is Auglab + v26_6_2 (train 50% prob), but this is with 0% synth data during validation. 
The second best is v26_6_2 (train 50% and val 100%). 

Next steps: 
1. confirm whether v26_6_2_50_100 > v26_6_2_train050_val000 with other folds. If yes then conclude training with synthetic validation data is superior. 
2. train Auglab + v26_6_2 with 50% train and 100%

**Comment by Luugaaa** [comment]
Synth val exp results: 

<img width="573" height="83" alt="Image" src="https://github.com/user-attachments/assets/3a8a97b9-b703-434f-9168-73e7208b1e9a" />

Conclusion: we should keep the synthetic data in validation 

Now, what's the impact of synth validation on auglab and auglab + synthseg EM?

**Comment by Luugaaa** [comment]
## Results on chaos (liver, spleen, kidney segmentation)

Note: CT is only evaluated on liver segmentation. 

### When training on T1n

<img width="1797" height="776" alt="Image" src="https://github.com/user-attachments/assets/d10b0d5a-4a1c-418d-ab0d-9ba4c68a9b55" />

<img width="1797" height="776" alt="Image" src="https://github.com/user-attachments/assets/61159ac2-989f-4b18-a3ad-2d4dc66d649a" />

We out perform all methods, we even beat the baseline In Domain. The results are very consistent accros contrasts. When applied alone, our method doesn't beat synthseg, the combination with auglab is not jsut beneficial but necessary. Ours + auglab consistently beats auglab, but not by a large margin, this alone could question the necessity of our approach. 


### When training on T2spir

<img width="1798" height="776" alt="Image" src="https://github.com/user-attachments/assets/0323d332-0886-49ae-a258-7fefbae5bba7" />

<img width="1798" height="776" alt="Image" src="https://github.com/user-attachments/assets/1b5a7717-ceb7-47f5-9a25-9f93787e5469" />

Here again we out perform all methods, even the baseline In Domain. Here, ours + auglab really outperforms auglab alone.

**Comment by Luugaaa** [comment]
Segmentation viz on T2SPIR: 
blue = GT
warm colors = pred

baseline: 

<img width="814" height="628" alt="Image" src="https://github.com/user-attachments/assets/c4a43cb9-0d44-42a2-bfd0-ff31aefd53ac" />

auglabAug + v26_6_2: 

<img width="814" height="628" alt="Image" src="https://github.com/user-attachments/assets/cbc9d301-7644-4d7c-ad66-d75dcc85fa43" />

There's only a ~10% dice score on the whole dataset, so it's hard to see a clear segmentation superiority on only one sample.

**Comment by Luugaaa** [comment]
<img width="2409" height="777" alt="Image" src="https://github.com/user-attachments/assets/ce4aa362-0f92-4a55-98b6-c72b5ae0c3f1" />

our v26_6_2
<img width="814" height="883" alt="Image" src="https://github.com/user-attachments/assets/1dc051f8-35e0-41ed-b75d-d7b9150e9550" />

auglab default (for example) -- lot of oevrsegmentation: 

<img width="714" height="724" alt="Image" src="https://github.com/user-attachments/assets/fbd6200f-421d-4d0d-9f92-f894e42c910c" />

**Comment by Luugaaa** [comment]
<img width="1798" height="776" alt="Image" src="https://github.com/user-attachments/assets/777a8497-085c-4777-a1a5-f937e5891100" />

In the table above (re-pasted in this comment for clarity), we can see the baseline performance is quite worse than other experiments' results. I found out it's due to a specific subject (MR36) that has some kind of abnormality. 

<img width="932" height="530" alt="Image" src="https://github.com/user-attachments/assets/292b3d20-0874-448a-b685-532ae26c0364" />

| Case | baseline | auglabAug | auglab_default |
| :--- | :--- | :--- | :--- |
| MR05 | 0.914 | 0.899 | 0.912 |
| MR19 | 0.925 | 0.907 | 0.915 |
| MR38 | 0.852 | 0.844 | 0.861 |
| MR36 | 0.508 | 0.828 | 0.773 |
| Overall | 0.800 | 0.870 | 0.865 |

This subject happens to be part of the test set... 
So it's not a code error, but a harder segmentation case on which only the baseline is failing. Is it acceptable to keep it this way @jcohenadad?

**Comment by jcohenadad** [comment]
> So it's not a code error, but a harder segmentation case on which only the baseline is failing. Is it acceptable to keep it this way @jcohenadad?

I'd say so, yes

**Comment by Luugaaa** [comment]
The main issue with AMOS was the orientation. It was RAS while everything else was LPS. It's not fixed. The updated results are below: 

T1in training: 
<img width="2409" height="777" alt="Image" src="https://github.com/user-attachments/assets/3625435c-2b1a-4fd0-9e4b-fca36de9e191" />

<img width="2415" height="777" alt="Image" src="https://github.com/user-attachments/assets/307f6d01-8f23-4e88-b52f-3e754975d853" />

t2spir training: 

<img width="2409" height="777" alt="Image" src="https://github.com/user-attachments/assets/66a21522-f84e-4c23-8c59-e245606f20a1" />

<img width="2416" height="777" alt="Image" src="https://github.com/user-attachments/assets/587a9497-d243-47ec-aa1f-919926d7395a" />

**Comment by Luugaaa** [comment]
There is still a big FOV issue, should we crop a box around the GT ? 

AMOS (bigger FOV -- test only dataset --> bad results)
<img width="782" height="729" alt="Image" src="https://github.com/user-attachments/assets/0b107ff2-5cb1-4540-b421-3b61fdaefaf1" />

CHAOS (smaller FOV -- training dataset --> good results)

<img width="782" height="558" alt="Image" src="https://github.com/user-attachments/assets/ac2e9d8a-3b0d-44dc-bc7d-53df4933f63d" />

**Comment by Luugaaa** [comment]
Added the CRIMRI dataset to the test set, the predictions don't make much sense, I suspect this is mostly due to the hypertrophic spleen (I believe this is linked to the cirrhotic condition). Won't keep it in the test set. Prediction example below (auglabAug_v26_6_2_train050_val000): 

<img width="621" height="502" alt="Image" src="https://github.com/user-attachments/assets/01a74c0a-48f2-485e-8b82-64db7bff8c9a" />

The model seems confused and fragmented both the spleen (left) and the liver (right) into different regions. Ruled out FOV, sampling and orientation issues after a thorough investigation. 


Follow-up: fully investigated, not just the hypertrophic-spleen guess above — see `CIRRHOSIS_FRAGMENTATION_INVESTIGATION.md`. Ruled out FOV, orientation, intensity/normalization, trainer-class code, patch-tiling, config, and resampling with direct tests. Leading explanation: cirrhosis → portal hypertension → splenomegaly, an anatomical distortion none of our contrast-only augmentations randomize. Formally excluded from the cross-dataset configs (2026-08-01); per-dataset results kept for reference, flagged untrustworthy.

**Comment by Luugaaa** [comment]
Excluded the kidney-t2w dataset too (https://zenodo.org/records/5153568), because a coronal FOV issue. These coronal acquisitions are too different from the chaos acquisition and do not provide enough context to the models. 

kidney-t2w: 
<img width="381" height="378" alt="Image" src="https://github.com/user-attachments/assets/baa72150-186a-43e8-b217-cf6040e3d168" />

chaos: 

<img width="381" height="378" alt="Image" src="https://github.com/user-attachments/assets/bd8024cc-7efd-40c2-8a45-e74ec02c0ba1" />

**Comment by Luugaaa** [comment]
FOV issue above is resolved. Built a FOV-restricted eval: for each anchor organ (kidney/liver/spleen) we measure CHAOS's own median S-I margin from training GT (`06_30_measure_chaos_fov.py` → `chaos_fov_margins.json`), then crop test volumes to the CHAOS-equivalent slab before scoring. Shared implementation: `00_commun_scripts/00_00_utils/fov.py`. This is what the current AMOS/SLIVER07 numbers below reflect — not documented in the paper draft yet, should add a sentence to setup/supplementary.

**Comment by Luugaaa** [comment]
Numbers above predate the AMOS orientation fix, FOV restriction, and the 2026-07-29 checkpoint fix. Current CHAOS column (T1in+T2spir pooled, AMOS+SLIVER07 included via FOV-restricted eval):

| config | Dice | HD95 (mm) |
|---|---|---|
| OURS · val000 | 86.3 | 28.8 |
| OURS · val100 | 86.6 | 27.5 |
| auglab_default (best Dice rival) | 85.2 | — |
| srcsm (best HD95 rival) | — | 28.3 |

Per-modality: T1in 87.4 Dice / 24.3 HD95, T2spir 87.2 / 24.3. Significant vs. all 5 competitors, p ≤ 1.8e-23. Source: `datasets/01_commun_results/meta_task_heatmap_summary.md`, paper `tab:meta`.

**Comment by Luugaaa** [comment]
Following the decision in issue #25 a `contrast_groups` mechanism pools CT into one vote (liver/spleen/kidney sub-averaged, then averaged against t1in/t2spir/t1out/t2w as equals) instead of counting liver-CT sources (chaos-own + amos_ct + sliver07) as separate flat columns that outvoted the single-source modalities. 

Also fixed: the table used to print CT's 3 raw sources as separate columns alongside a differently-pooled `all`, which read as inconsistent/ungrouped -- and AMOS's MRI column was named `amos_mri` (a dataset label, not a contrast). Renamed to `t2w` and collapsed the table to one column per real contrast + one CT column:

| method | t1in | t2spir | t1out | t2w | CT | all | sig. vs ref |
|---|---|---|---|---|---|---|---|
| baseline | 42.8 | 41.3 | 31.9 | 22.0 | 8.2 | 29.3 | 1.7e-06 |
| synthseg_noEM | 48.0 | 78.6 | 52.4 | 69.8 | 49.3 | 59.6 | 1.8e-06 |
| synthseg_EM | 82.0 | 87.4 | 83.9 | 86.8 | 74.9 | 83.0 | 0.005 |
| auglab_default | 83.5 | 87.1 | 84.4 | 87.4 | 74.1 | 83.3 | 0.023 |
| srcsm | 84.8 | 86.7 | **86.4** | **87.5** | 73.8 | 83.9 | 0.246 |
| **ours (train050_val000)** | 85.0 | 88.3 | 85.6 | 87.3 | 78.6 | 85.0 | -- |
| ours (train050_val100) | **85.8** | **88.4** | 85.7 | 86.9 | **79.4** | **85.2** | 0.917 |

HD95 mm (down):

| method | t1in | t2spir | t1out | t2w | CT | all | sig. vs ref |
|---|---|---|---|---|---|---|---|
| baseline | 91.6 | 107.3 | 120.0 | 137.7 | 165.4 | 124.4 | 9.5e-07 |
| synthseg_noEM | 105.4 | 78.2 | 83.8 | 64.3 | 117.9 | 89.9 | 8.3e-07 |
| synthseg_EM | 25.7 | 29.8 | 23.3 | 21.1 | 59.1 | 31.8 | 0.453 |
| auglab_default | 24.5 | 40.7 | 25.0 | 22.5 | 83.7 | 39.3 | 0.040 |
| srcsm | 20.4 | **25.1** | **15.0** | **17.1** | 56.3 | **26.7** | 1.000 |
| **ours (train050_val000)** | 22.7 | 30.4 | 22.0 | 21.2 | 56.7 | 30.6 | -- |
| ours (train050_val100) | **16.8** | 31.6 | 17.5 | 21.5 | **56.2** | 28.7 | 1.000 |

Note the CT column dropped noticeably from the old flat display (e.g. baseline's raw `ct` used to read 21.7 -- now the properly-pooled CT column reads 8.2, since the flat column was hiding how badly baseline does on spleen/kidney once those count instead of just liver). Still significant vs. all methods except srcsm (Dice p=0.246, HD95 p=1.000 -- srcsm remains the closest rival, best HD95 outright at 26.7mm).
